test/tests/amqp/test_amqp_racknode.py

```diff
@@ -0,0 +1,166 @@
+'''
+Copyright 2017 Dell Inc. or its subsidiaries.  All Rights Reserved.
+
+Author(s):
+Norton Luo
+
+'''
+import json
+from time import sleep
+import os
+import sys
+import json
+import time
+import string
+from datetime import *
+import time
+import subprocess
+# set path to common libraries
+sys.path.append(subprocess.check_output("git rev-parse --show-toplevel", shell=True).rstrip("\n") + "/test/common")
+import fit_common
+import test_api_utils
+import fit_amqp
+
+amqp_message_received = False
+routingkey = ""
+amqpbody = ""
+
+
+def get_node_list_by_type(type):
+    mon_url = '/api/2.0/nodes'
+    response = fit_common.rackhdapi(mon_url)
+    nodefound = []
+    if response['status'] in range(200, 205):
+        for nodes in response['json']:
+            if nodes["type"] == type:
+                nodefound.append(nodes["id"])
+    if fit_common.VERBOSITY >= 2:
+        print type," list=", nodefound
+    return nodefound
+
+
+def amqpcallback(ch, method, properties, body):
+    if fit_common.VERBOSITY >= 2:
+        print(" Routing Key %r:" % (method.routing_key))
+        print body
+    global amqp_message_received
+    global routingkey, amqpbody
+    amqp_message_received = True
+    amqpbody = body
+    routingkey = method.routing_key
+
+
+# Select nose.plugins.attrib import attr
+from nose.plugins.attrib import attr
+
+
+@attr(all=True, regression=True, smoke=True)
+class amqp_rack_node(fit_common.unittest.TestCase):
+    # clear the test environment
+    def tear_down(self):
+        self.test_api_delete_rack()
+
+    def compare_message(self, expectedkey, expectedpayload):
+        global routingkey, amqpbody
+        self.assertEquals(routingkey, expectedkey, "No AMQP message received")
+        try:
+            amqp_body_json = fit_common.json.loads(amqpbody)
+        except:
+            print "FAILURE - The message body is not json format! "
+            return False
+        try:
+            self.assertEquals(amqp_body_json['version'], expectedpayload['version'], "version field not correct!")

```

Comment:
same as above assert comments for these (the unittest asserts don't print the values for you)